gsa/src/web/components/powerfilter/severityvaluesgroup.js

```diff
@@ -16,64 +16,41 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
-import React from 'react';
+import React, {useState} from 'react';
 
 import PropTypes from 'web/utils/proptypes.js';
 
 import FormGroup from 'web/components/form/formgroup.js';
 import {parseSeverity} from 'gmp/parser.js';
 import RelationSelector from 'web/components/powerfilter/relationselector';
 import NumberField from 'web/components/form/numberfield';
-import Divider from '../layout/divider';
-
-class SeverityValuesGroup extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      relation: '=',
-    };
-
-    this.handleRelationChange = this.handleRelationChange.bind(this);
-  }
-
-  handleRelationChange(rel) {
-    this.setState({
-      relation: rel,
-    });
-    const severity = parseSeverity(this.props.filter.get(this.props.name));
-    const keyword = this.props.name;
-
-    this.props.onChange(severity, keyword, rel);
-  }
-
-  render() {
-    const severity = parseSeverity(this.props.filter.get(this.props.name));
-    const newRelation = this.state.relation;
-    const keyword = this.props.name;
-    return (
-      <FormGroup title={this.props.title}>
-        <Divider>
-          <RelationSelector
-            relation={newRelation}
-            onChange={this.handleRelationChange}
-          />
-          <NumberField
-            name={this.props.name}
-            type="int"
-            min={0}
-            max={10}
-            value={severity}
-            size="5"
-            onChange={(value = severity, name = keyword) =>
-              this.props.onChange(value, name, newRelation)
-            }
-          />
-        </Divider>
-      </FormGroup>
-    );
-  }
-}
+import Divider from 'web/components/layout/divider';
+
+const SeverityValuesGroup = ({filter, name, title, onChange}) => {
+  const [rel, setRel] = useState('=');
+  const severity = parseSeverity(filter.get(name));
+  const keyword = name;
+
+  return (
+    <FormGroup title={title}>
+      <Divider>
+        <RelationSelector relation={rel} onChange={newRel => setRel(newRel)} />
```

Comment:
I am seeing a potential bug here. Changing the relation isn't propagated to the calling component anymore. You should call the onChange handler here too. Something like
```jsx
<RelationSelector relation={rel} onChange={newRel => {setRel(newRel); onChange(severity, name, newRel);} />
```

Reply:
thanks. done.